File: Frog jump with k distance/Tabulation.java
import java.util.Arrays;
public class Tabulation {
       public static void main(String[] args) {
        int arr[] = {10, 5, 20, 0, 15};
        int size = arr.length;
        int k = 2;
        int[] dp = new int[size];
        dp[0] = 0;

        for(int i = 1; i < size; i++ ) {
            int minimumJump = Integer.MAX_VALUE;

            for(int j = 1; j <= k ;j ++) {
                if(i - j >= 0) 
                    minimumJump = Math.min(minimumJump,dp[i-j] + Math.abs(arr[i] -arr[i-j]));
                else break;
            }

            dp[i] = minimumJump;
        }
        System.out.println(Arrays.toString(dp));
    }
}

File: Frog jump with k distance/Tabulation.py
def FrogJump():
    arr = [10, 5, 20, 0, 15]
    size = len(arr)-1
    k = 2
    dp = [-1 for _ in range(size+1)]
    dp[0] = 0

    for i in range(1,size+1):
        minimum = float('inf')

        for j in range(1,k+1):
            if i - j >= 0: minimum = min(minimum,dp[i-j] + abs(arr[i] - arr[i-j]))
            else: break 
        dp[i] = minimum

    return dp[size]

if __name__ == "__main__":
    print(FrogJump())